### ISSUE TITLE
stability:  20k+ goroutines deadlocked on beginCmd, no more progress

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#5368]
(to be updated shortly; want an issue number)

**Comment by tschottdorf** [comment]
Tasks running on the offending node:

```
40539  server/node.go:701
1      storage/queue.go:380
```

Somehow all servers and clients are happy (regular leader lease activity as well), but nothing is really going forward. Losing a Raft command could explain some of that (blocking all client commands, but not blocking internal stuff), I suppose.

```
grep -c beginCmds *.goroutine
big_guy.ec2-54-208-199-194.compute-1.amazonaws.com.goroutine:24848
ec2-54-209-133-121.compute-1.amazonaws.com.goroutine:1
ec2-54-209-150-36.compute-1.amazonaws.com.goroutine:0
ec2-54-209-69-52.compute-1.amazonaws.com.goroutine:2
```

logs copied and data copied to `logs.5368` on the server. Goroutines etc attached.
[Archive.zip](https://github.com/cockroachdb/cockroach/files/179036/Archive.zip)

**Comment by tschottdorf** [comment]
the UI sometimes gets its data, but then cuts out irregularly. So it's not completely happy. Wonder what's causing that - I would've expected it to work or not, but nothing in between.

**Comment by tschottdorf** [comment]
```
ubuntu@ip-172-31-58-174:~$ ./cockroach debug range --ca-cert certs/ca.crt --key certs/root.client.key --cert certs/root.client.crt ls                                                │ubuntu@ip-172-31-58-173:~$ ./cockroach debug range --ca-cert certs/ca.crt --key certs/root.client.key --cert certs/root.client.crt ls
Error: scan failed: client/rpc_sender.go:58: roachpb.Batch RPC failed: context deadline exceeded                                                                                     │Error: scan failed: client/rpc_sender.go:58: roachpb.Batch RPC failed: context deadline exceeded
                                                                                                                                                                                     │
Failed running "debug"
```

**Comment by bdarnell** [comment]
Fixed by #5551, right?
